salt/_pillar/metalk8s_endpoints.py

```diff
@@ -16,6 +16,48 @@ def __virtual__():
         return __virtualname__
 
 
+def namespaced_service_endpoints_nodeport(service, namespace, kubeconfig):
+    endpoint = namespaced_service_endpoints(service, namespace, kubeconfig)
+    node_name = endpoint.get('node_name')
+
+    if not node_name:
+        log.exception('Cannot get `node_name` from %s.', endpoint)
```

Comment:
why `log.exception`, there is no exception raised here?

Reply:
Yes, `log.error` would be better. Even then, we should start implementing the `_errors` key as described in #1018